pkg/router/template/router.go

```diff
@@ -450,6 +451,8 @@ func (r *templateRouter) AddRoute(id string, route *routeapi.Route, host string)
 		}
 	}
 
+	config.RoutingKeyName = fmt.Sprintf("%x", md5.Sum([]byte(fmt.Sprintf("%s %s", config.TLSTermination, backendKey))))
```

Comment:
The use of tlsterm and route key (namespace + route name) maintains the current semantics such that if the type or name of a route changes, it's cookie id will differ accordingly.  Without a salt brute force recovery would be possible, but I'm not sure that's a problem.